### PR TITLE
feat: Add new helper methods

### PR DESCRIPTION
**Pull request by sillyElvy**
## About
Adds getting dm channel and modifying nick name of the bot in guilds
## Checklist

- [x] The ``pre-commit`` code linter has been run over all edited files to ensure the code is linted.
- [x] I've ensured the change(s) work on `3.8.6` and higher.

I've made this pull request: (check all that apply)
  - [ ] For the documentation
  - [x] To add a new feature
  - [ ] As a general enhancement
  - [ ] As a refactor of the library/the library's code
  - [ ] To fix an existing bug
  - [ ] To resolve #ISSUENUMBER

This is:
  - [ ] A breaking change

### CODE REVIEW

File: interactions/api/http/user.py [hunk 1]
```diff
@@ -58,7 +58,9 @@ async def modify_self_nick_in_guild(self, guild_id: int, nickname: Optional[str]
         :return: Nothing needed to be yielded.
         """
         return await self._req.request(
-            Route("PATCH", "/guilds/{guild_id}/members/@me/nick", guild_id=guild_id),
+            Route(
+                "PATCH", "/guilds/{guild_id}/members/@me", guild_id=guild_id
+            ),  # /nick isdeprecated
```
Comment (sillyElvy) [inline interactions/api/http/user.py]:
```suggestion
            ),  # /nick is deprecated
```

File: interactions/api/models/guild.py [hunk 2]
```diff
@@ -2930,6 +2930,36 @@ async def get_invites(self) -> List["Invite"]:
...
+
+        _member = Member(
+            **await self._client.modify_self_nick_in_guild(int(self.id), new_nick),
+            _client=self._client,
+        )
+        _member_id = int(_member.id)
+        if self.members is None:
+            self.members = []
+        for index, member in enumerate(self.members):
+            if int(member.id) == _member_id:
```
Comment (Damego) [inline interactions/api/models/guild.py:2959]: You know `Snowflake` has `__eq__` so you can remove `int()` here and above

Comment (sillyElvy) [reply interactions/api/models/guild.py:2959]: It's the same in the whole guild file so it's out of the scope of this PR

Comment (Damego) [reply interactions/api/models/guild.py:2959]: fine

File: interactions/client/bot.py [hunk 3]
```diff
@@ -155,6 +155,28 @@ def __init__(
...
+
+        Sets a new nick in the specified guild.
+
+        :param Union[int, str, Snowflake, Guild] guild_id: The ID of the guild to modify the nick in
+        :param Optional[str] new_nick: The new nick to assign
+        """
+        if not self._http or isinstance(self._http, str):
+            raise LibraryException(
+                code=13, message="You cannot use this method until the bot has started!"
+            )
```
Comment (Damego) [inline interactions/client/bot.py:172]:
```suggestion
            )

```

File: interactions/client/bot.py [hunk 4]
```diff
@@ -155,6 +155,28 @@ def __init__(
...
+        if new_nick is MISSING:
+            raise LibraryException(code=12, message="new nick name must either a string or `None`")
```
Comment (Damego) [inline interactions/client/bot.py:175]:
```suggestion
            raise LibraryException(code=12, message="new nick name must either a string or `None`")

```

File: interactions/client/bot.py [hunk 5]
```diff
@@ -155,6 +155,28 @@ def __init__(
...
+        _id = int(guild_id.id) if isinstance(guild_id, Guild) else int(guild_id)
```
Comment (Damego) [inline interactions/client/bot.py:177]:
```suggestion
        _id = int(guild_id.id) if isinstance(guild_id, Guild) else int(guild_id)

```

File: interactions/client/bot.py [hunk 6]
```diff
@@ -1517,6 +1539,10 @@ async def modify(
         :return: The modified User object
         :rtype: User
         """
+        if not self._http or isinstance(self._http, str):
+            raise LibraryException(
+                code=13, message="You cannot use this method until the bot has started!"
+            )
```
Comment (Damego) [inline interactions/client/bot.py:1548]:
```suggestion
            )

```

File: interactions/api/models/user.py [hunk 7]
```diff
@@ -202,3 +203,18 @@ async def send(
...
+        """
+        .. versionadded:: 4.4.0
+
+        Gets the DM channel with the user
+
+        :return: The DM channel with the user
+        :rtype: Channel
+        """
+        if not self._client:
+            raise LibraryException(code=13)
```
Comment (Damego) [inline interactions/api/models/user.py:217]:
```suggestion
            raise LibraryException(code=13)

```

File: interactions/api/models/guild.py [hunk 8]
```diff
@@ -2930,6 +2930,36 @@ async def get_invites(self) -> List["Invite"]:
...
+        .. versionadded:: 4.4.0
+
+        Changes a nickname of the current bot user in a guild.
+
+        :param Optional[str] new_nick: The new nickname, if any.
+        :return: The modified bot member object
+        :rtype: Member
+        """
+        if not self._client:
+            raise LibraryException(code=13)
```
Comment (Damego) [inline interactions/api/models/guild.py:2944]:
```suggestion
            raise LibraryException(code=13)

```

**Review by Damego — COMMENTED** [review]
Adding blank lines to improve readability

**Review by Damego — APPROVED** [review]

**Pull request merged**